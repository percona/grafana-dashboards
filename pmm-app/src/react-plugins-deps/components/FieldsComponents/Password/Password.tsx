import React from 'react';
import { useField } from 'react-final-form-hooks';

export const PasswordField = ({ name, placeholder, required, form, ...rest }) => {
  // @ts-ignore
  const field = useField(name, form);
  return <input type={'password'} {...field.input} placeholder={placeholder} className="input-field input-field--dark" />;
};
