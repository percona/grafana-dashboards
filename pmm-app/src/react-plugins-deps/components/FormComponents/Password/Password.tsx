import React from 'react';
import { useField } from 'react-final-form';

export const PasswordField = ({ name, placeholder = '', required }) => {
  // @ts-ignore
  const { input } = useField(name);
  return (
    <input
      type="password"
      {...input}
      required={required}
      placeholder={`${required ? '*' : ''}${placeholder}`}
      className="input-field input-field--dark"
    />
  );
};
