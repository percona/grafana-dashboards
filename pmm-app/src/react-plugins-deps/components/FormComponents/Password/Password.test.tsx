// Just a stub test
import { PasswordField } from './Password';
import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from 'react-final-form';

describe('Password field test', () => {
  it('Password renders correct without props', () => {
    const component = renderer.create(
      <Form onSubmit={() => null} render={() => <PasswordField name="test_field" />} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
