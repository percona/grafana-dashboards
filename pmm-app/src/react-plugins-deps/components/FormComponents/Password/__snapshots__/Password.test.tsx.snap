// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password field test Password renders correct without props 1`] = `
<input
  className="input-field input-field--dark"
  name="test_field"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  placeholder=""
  type="password"
  value=""
/>
`;
